sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2034,4 +2034,25 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       }
     }
   }
+
+  test("SPARK-21677: json_tuple throws NullPointException when column is null as string type") {
```

Comment:
This is just an end-to-end test case. We also need to add unit test cases in `JsonExpressionsSuite`